### PR TITLE
fix(telemetry): add event data for step indicator component

### PR DESCRIPTION
**Pull request by jyasveer**
- modified data sent with the step click event
- broadcast event on click of project input in step indicator

### CODE REVIEW

File: src/app/launcher/shared/telemetry.decorator.ts [hunk 1]
```diff
@@ -26,8 +26,13 @@ export function broadcast(event: string, properties: any): MethodDecorator {
...
 
             const broadcast: Broadcaster = injectorInstance.get(Broadcaster);
-            const values = new PropertiesGetter(this).mapKeys(_.cloneDeep(properties));
+            let values = new PropertiesGetter(this).mapKeys(_.cloneDeep(properties));
 
+            if (event === 'stepIndicatorClicked') {
+                values = {
+                    step: args[0] ? args[0] : ''
+                };
+            }
```
Comment (edewit) [inline src/app/launcher/shared/telemetry.decorator.ts]: this seems very specific to one case, why would this be needed?

File: src/app/launcher/step-indicator/step-indicator.component.ts [hunk 2]
```diff
@@ -82,6 +82,9 @@ export class StepIndicatorComponent implements OnInit {
     this.launcherComponent.validateProjectName();
   }
 
+  @broadcast('stepIndicatorProjectInputClicked', {})
```
Comment (edewit) [inline src/app/launcher/step-indicator/step-indicator.component.ts]: shouldn't it be `stepIndicatorClicked`?

File: src/app/launcher/shared/telemetry.decorator.ts (hunk 1 above)
Comment (jyasveer) [reply src/app/launcher/shared/telemetry.decorator.ts]: For this event here, i get the data from the original method's argument. so I had to add the specific thing here.

File: src/app/launcher/step-indicator/step-indicator.component.ts (hunk 2 above)
Comment (jyasveer) [reply src/app/launcher/step-indicator/step-indicator.component.ts]: No what I have here is different from `stepIndicatorClicked`. This event is to understand if the users are really using the project name input box in the left navigation

**Comment by edewit** [comment]
superseded by #304

**Pull request closed without merging**